src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1628,6 +1628,15 @@ private static BoundBlock BindMethodBody(MethodSymbol method, TypeCompilationSta
                 body = null;
             }
 
+            // Delegate constructors and invokes have no body
+            if (method.IsExtern ||
+                method.MethodKind == MethodKind.DelegateInvoke ||
+                (method.MethodKind == MethodKind.Constructor &&
+                method.ContainingType?.IsDelegateType() == true))
```

Comment:
> method.ContainingType?.IsDelegateType() == true [](start = 15, length = 48)

I think the method kind checks are redundant, the check whether containing type is a delegate type should cover both of them, the same way this is done in the assert below

Reply:
I think you are right. Just updated it.